docs/docs/examples.md

```diff
@@ -0,0 +1,317 @@
+3. Line 8 execution successfully guarantees that the second child is indeed an ASN.1 OCTET STRING encapsulating
+another ASN.1 structure.
+4. Lines 9-10 verify that the first child contained in the ASN.1 OCTET STRING
+    * is an ASN.1 primitive
+    * tagged with `TAG_TIME_RELATIVE`
+    * containing an ASN.1 boolean
+5. Line 12 ensures that the next child is an ASN.1 primitive, encoding an unsigned integer (in case an `UInt` is expected)
+6. Line 13 tackles the alternative and ensures that the next child contains a properly encoded ASN.1 time
+7. Lastly, lines 15-16 make sure no additional content is present, thus fully verifying the structure as a whole.
+
+
+## Issuing Binding Certificates
+
+!!! info  inline end
+    This example requires the _Supreme_ KMP crypto provider. Only _Signum_-specifics are illustrated using code snippets.
+
+We'll assume a JVM backend using _[WARDEN](https://github.com/a-sit-plus/warden)_ and trust anchors all set up correctly
+on the client apps.
+A common pattern in a mobile client setting in the context of banking or eID are so-called _binding certificates_ (or binding keys, but we'll stick to certificates here).
+Just assume a bank with a mobile client application: Customers are typically issued an activation token out-of-band
+(via mail, by the teller, …). This token is used to activate the app and transactions can then be authorized using biometrics.
+
+In settings as critical as eID and banking, the service operator typically wants to ensure that only uncompromised clients
+may access a service. To ensure this, the example described here relies on attestation.
+
+This process works more or less as follows:
+
+1. The client contacts the back-end to start the binding process
+2. The back-end authenticates the binding request, identifying the customer. This could be a traditional authentication process, some out-of-band personalized token, etc.
+3. The user enters this information into the client app and the app transmits this information to the back-end.
+4. The back-end sends a challenge to the client
+5. The client creates a new public-private key pair, using the challenge to also attest app, key, and the biometric authorization requirement (see [Attestation](supreme.md#attestation)).
+   ```kotlin
+   val signer = PlatformSigningProvider.createSigningKey("binding") {
+     ec { curve = ECCurve.SECP_256_R_1 }
+     hardware {
+        backing = REQUIRED
+        attestation { challenge = challengeFromServer }
+        protection {
+            factors { biometry = true }
+        }
+     }
+   }.getOrElse { TODO("Handle error") }
+   ```
+6. The client creates and signs a CSR for the key, which includes the challenge and an attestation proof
+      ```kotlin
+      val tbsCSR = TbsCertificationRequest(
+          subjectName = listOf(RelativeDistinguishedName(AttributeTypeAndValue.CommonName(Asn1String.UTF8("client")))),
```

Comment:
note this down as one of the things that should definitely have a convenience helper

Reply:
#183